### DISCUSSION TITLE
swell-js filters not returing filterable but hidden attributes

### DISCUSSION THREAD

**Discussion by siphosenkosindhlovu** [swellstores/community#53] [Help] [answered]
I'm not quite sure on the intended behaviour, but when getting product filters using the Frontend API through swell-js, some attributes aren't returned if they are marked and hidden and filterable and attributes marked as visible but not filterable still show up on the filter list.

**Comment by logeshswell** [comment, marked as answer]
@siphosenkosindhlovu Let me help you with the expected behavior here.
* Only Visible attributes will be returned in swell-js to show up on the storefront.
  * Working as expected.
* Only Filterable attributes should be returned in the swell-js product filters.
  * Not working as expected. I filed a bug ticket already and let me keep you posted once it is fixed.
  
At the moment, filterable toggle doesn't have any impact and so it is recommended to enable/disable either flags together to show/hide in the product details and filters API.

Does this answer your question? If not, please reply to this thread and I am happy to answer you.
